src/Microsoft.ML.Core/CommandLine/ArgumentAttribute.cs

```diff
@@ -139,5 +140,11 @@ public bool IsRequired
         {
             get { return ArgumentType.Required == (_type & ArgumentType.Required); }
         }
+
+        public Type SignatureType
```

Comment:
>SignatureType [](start = 20, length = 13)

Does it have to be here? Let me see why it would be needed

Reply:
Ooh, I see now. You need to tell the arg parser what's the signature type to invoke the ctor properly. 
Hmm, this is unfortunate. Are you sure there's no better way? 

---
In reply to: [208054268](https://github.com/dotnet/machinelearning/pull/622#discussion_r208054268) [](ancestors = 208054268)

Reply:
I haven't found one yet...

We could have some sort of mapping from `IComponentFactory<...>` to `SignatureType`. But that would seem fragile to me.

My other thinking (that seems to be aligned with @TomFinley's above) is that we should be moving away from `SignatureTypes`.  This at least puts the Signature type into the attribute, and out of the API directly.  Hopefully, eventually we can remove signature types all together.

Reply:
Yea, the fact that it's prominent in the attribute makes the whole approach less desirable. But I think it's an OK intermediate step

---
In reply to: [208058930](https://github.com/dotnet/machinelearning/pull/622#discussion_r208058930) [](ancestors = 208058930)